apps/agreement/contracts/disputable/DisputableApp.sol

```diff
@@ -0,0 +1,177 @@
+    * @dev IForwarder interface conformance
+    * @param _sender Address of the account intending to forward an action
+    * @return True if the given address can submit actions, false otherwise
+    */
+    function canForward(address _sender, bytes) public view returns (bool);
+
+    /**
+    * @dev Create a new action in the agreement
+    * @param _disputableId Identification number of the disputable action in the context of the disputable
+    * @param _submitter Address of the user that has submitted the action
+    * @param _context Link to a human-readable text giving context for the given action
+    * @return Unique identification number for the created action in the context of the agreement
+    */
+    function _newAction(uint256 _disputableId, address _submitter, bytes _context) internal returns (uint256) {
+        IAgreement agreementApp = _getAgreement();
+        require(agreementApp != IAgreement(0), ERROR_AGREEMENT_NOT_SET);
+        return agreementApp.newAction(_disputableId, _submitter, _context);
+    }
+
+    /**
+    * @dev Close action in the agreement
+    * @param _actionId Identification number of the disputable action in the context of the agreement
+    */
+    function _closeAction(uint256 _actionId) internal {
+        IAgreement agreementApp = _getAgreement();
+        require(agreementApp != IAgreement(0), ERROR_AGREEMENT_NOT_SET);
+        agreementApp.closeAction(_actionId);
+    }
+
+    /**
+    * @dev Reject disputable
+    * @param _disputableId Identification number of the disputable to be rejected
+    */
+    function _onDisputableRejected(uint256 _disputableId) internal;
+
+    /**
+    * @dev Challenge disputable
+    * @param _disputableId Identification number of the disputable to be challenged
+    * @param _challenger Address challenging the disputable
+    */
+    function _onDisputableChallenged(uint256 _disputableId, address _challenger) internal;
+
+    /**
+    * @dev Allow disputable
+    * @param _disputableId Identification number of the disputable to be allowed
+    */
+    function _onDisputableAllowed(uint256 _disputableId) internal;
+
+    /**
+    * @dev Void disputable
+    * @param _disputableId Identification number of the disputable to be voided
+    */
+    function _onDisputableVoided(uint256 _disputableId) internal;
+
+    /**
+    * @dev Tell the agreement linked to the disputable instance
+    * @return Agreement linked to the disputable instance
+    */
+    function _getAgreement() internal view returns (IAgreement) {
+        return IAgreement(AGREEMENT_POSITION.getStorageAddress());
+    }
+
+    /**
+    * @dev Tell whether an action can proceed or not, i.e. if its not being challenged or disputed
+    * @param _actionId Identification number of the action being queried in the context of the Agreement app
+    * @return True if the action can proceed, false otherwise
+    */
+    function _canProceed(uint256 _actionId) internal view returns (bool) {
```

Comment:
I guess this will change for each disputable app logic. Is it worth it to implement this generic version here? Same goes for `_closeAction`.

Reply:
I think it is a way to provide a call helper for both functionalities, it shouldn't change the implementation based on the disputable calling these don't you think?